src/core/stream_send.c

```diff
@@ -1531,16 +1531,14 @@ QuicStreamOnAck(
         Stream->Flags.LocalCloseResetReliableAcked &&
         Stream->UnAckedOffset >= Stream->ReliableOffsetSend;
     if (ReliableResetShutdown) {
-        Stream->Flags.LocalCloseAcked = TRUE;
-        Stream->Flags.RemoteCloseAcked = TRUE;
         QuicTraceEvent(
             StreamSendState,
             "[strm][%p] Send State: %hhu",
             Stream,
             QuicStreamSendGetState(Stream));
         QuicStreamCleanupReliableReset(Stream);
         Stream->Flags.LocalCloseReset = TRUE;
-        QuicStreamSendShutdown(Stream, FALSE, TRUE, FALSE, 0x21);
+        QuicStreamSendShutdown(Stream, FALSE, TRUE, FALSE, Stream->SendShutdownErrorCode);
```

Comment:
You should update the tests to use different error codes for different scenarios and ensure the proper ones are actually received by the peer.

Reply:
Added tests AND a new Event to alert the app that the stream has been aborted reliably for extra clarity.